### PR TITLE
Rollout custom ad notifcations

### PR DESCRIPTION
**Pull request by tmancey**
Windows 10 build 16299 and older or if native notifications are disabled
macOS 10.13 and older
Android 7 and older
All versions of Linux

### CODE REVIEW

**Comment by btlechowski** [comment]
This PR fixes https://github.com/brave/brave-browser/issues/18342

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers. QA has verified the PR/change on `main` (staging) via https://github.com/brave/brave-variations/pull/120#issuecomment-926063472.

**Pull request merged**
